Update supermarket-omnibus-cookbook to use chef-ingredient

irvingpop opened the pull request:
Finishing the work of @binamov,  this completes the work to convert to chef-ingredient with working tests.

New attributes:
`default['supermarket_omnibus']['package_version']` can be a Semver like `1.10.0` or the default `:latest`

Note some breaking changes, these will require a major version bump:
* `default['supermarket_package']['packagecloud_repo']` -> `default['supermarket_omnibus']['package_repo']`
* the repos have changed their name from `chef/stable` to `chef-stable` or `chef-current`

The new way to override the `yum-chef` or `apt-chef` repo location is by overriding the attributes specified in those cookbooks.

Fixes: #11 #15 #16

CC:  @jtimberman @nellshamrell @lauck for review

jtimberman commented:
Note that the attribute overriding in `yum-chef` and `apt-chef` is certainly not ideal since the attributes are named differently (so they're in parity with yum or apt config options, not because of arbitrary reasons). @sersut and I are working on ways to make that better in either `chef_ingredient`, or add that logic/ functionality to `mixlib-install`, but this is where we are now.

jtimberman commented on spec/unit/recipes/default_spec.rb:
```diff
@@ -5,18 +5,19 @@
 # Copyright (c) 2015 Irving Popovetsky, All Rights Reserved.
 
 require 'spec_helper'
+require 'mixlib/versioning' # new chef-ingredient makes heavy use of this library
```
It seems weird to me that you have to require this here. Is it not getting required in the right place in `chef_ingredient`? Does there need to be a Gemfile with this included?

jtimberman replied:
@schisamo @sersut Perhaps it's worth considering leveraging `mixlib-versioning` for version-related matters in Chef, in general, and including it in Chef/ChefDK omnibus packages?

irvingpop replied:
Good point @jtimberman -  I added that line after discovering that my ChefDK didn't include mixlib-versionining and the version I had installed was too old (1.0.0) and kept throwing this error:

```
    NoMethodError:
       chef_ingredient[supermarket] (supermarket-omnibus-cookbook::default line 56) had an error: NoMethodError: undefined method `major' for "0.0.0":String
     # /Users/irving/.chefdk/gem/ruby/2.1.0/gems/mixlib-versioning-1.0.0/lib/mixlib/versioning/format.rb:203:in `<=>'
     # /var/folders/bw/nf13rmz110bf83rqf6658mxm0000gq/T/d20150914-17034-ve4uaq/cookbooks/chef-ingredient/libraries/helpers.rb:49:in `>'
     # /var/folders/bw/nf13rmz110bf83rqf6658mxm0000gq/T/d20150914-17034-ve4uaq/cookbooks/chef-ingredient/libraries/helpers.rb:49:in `block in package_resource'
     # /var/folders/bw/nf13rmz110bf83rqf6658mxm0000gq/T/d20150914-17034-ve4uaq/cookbooks/chef-ingredient/libraries/helpers.rb:46:in `package_resource'
     # /var/folders/bw/nf13rmz110bf83rqf6658mxm0000gq/T/d20150914-17034-ve4uaq/cookbooks/chef-ingredient/libraries/chef_ingredient_provider.rb:40:in `block in <class:ChefIngredient>'
     # ./spec/unit/recipes/default_spec.rb:114:in `block (3 levels) in <top (required)>'
     # ./spec/unit/recipes/default_spec.rb:122:in `block (3 levels) in <top (required)>'
```

The comment should probably have been clearer about the version required.

schisamo replied:
This is how we worked around a similar issue for `chef-sugar`:
https://github.com/chef-cookbooks/omnibus/blob/master/.travis.yml#L21

You could obviously also include a `Gemfile` in the repo but in the case above we wanted to avoid introducing a Gemfile that might cause confusion.

irvingpop commented:
Thanks @jtimberman - I added travis-ci integration [here](https://github.com/chef-cookbooks/supermarket-omnibus-cookbook/commit/2e85c6cd73860656ddb18b91f94b915b32976fbf) and removed the unneeded require, everything is still green.

I also manually ran `kitchen test` on CentOS 6 and Ubuntu 14.04 and both passed.   Will work on kitchen testing in travis in a later commit.
